code/game/gamemodes/events/powercreeper.dm

```diff
@@ -110,28 +102,40 @@
 /obj/structure/cable/powercreeper/proc/die()
 	grown = 0 //we can't attack or spread anymore
 	do_flick(src, "death[add_state]", 13)
+	updateNeighbours(TRUE)
 	qdel(src)
 
 /obj/structure/cable/powercreeper/proc/try_electrocution(var/mob/living/M)
 	if(!istype(M))
 		return 0
 	playsound(src,'sound/weapons/electriczap.ogg',50, 1) //we still want a sound
-	if(!electrocute_mob(M, powernet, src))
-		M.apply_damage((powernet.avail / 3000), BURN) //one burn damage per 6 plants (keep in mind, this will also take the power from any connected powernet)
+	return electrocute_mob(M, powernet, src)
+
+/obj/structure/cable/powercreeper/proc/getViableNeighbours()
+	for(var/dir in cardinal)
+		var/turf/T = get_step(src, dir)
+		if(isViableGrow(T))
+			growdirs |= dir
+
+/obj/structure/cable/powercreeper/proc/isViableGrow(var/turf/T)
+	if(!T.has_gravity())
+		return 0
+	if(!T.Adjacent(src))
+		return 0
+	if(locate(/obj/structure/cable/powercreeper) in T)
+		return 0
+	if((T.density == 1) || T.has_dense_content())
 		return 0
 	return 1
 
-/obj/structure/cable/powercreeper/proc/getViableNeighbours()
-	. = list()
-	var/turf/T
+/obj/structure/cable/powercreeper/proc/updateNeighbours(var/dying = FALSE)
```

Comment:
this will only free up tiles if a creeper dies, not if the density changes. this means for example doors would permanently block a creepers growth.

I don't like it

Reply:
Working on it

Reply:
Sorted now.